x/dualstaking/keeper/delegator_reward.go

```diff
@@ -209,7 +209,7 @@ func (k Keeper) RewardProvidersAndDelegators(ctx sdk.Context, provider string, c
 
 	relevantDelegations := lavaslices.Filter(delegations,
 		func(d types.Delegation) bool {
-			return d.ChainID == chainID && d.IsFirstMonthPassed(ctx.BlockTime().UTC().Unix()) && d.Delegator != d.Provider
+			return d.ChainID == chainID && d.IsFirstMonthPassed(ctx.BlockTime().UTC().Unix()) && (d.Delegator != stakeEntry.Vault && d.Provider == stakeEntry.Operator)
```

Comment:
why isnt d.Delegator != stakeEntry.Vault enough?

Reply:
the other check was some sort of sanity check, but it's really not needed. Removed https://github.com/lavanet/lava/commit/323d4127c0e62f75fb078c4dc5eda7478a8abaa8